sfn/src/main/scala/uk/gov/nationalarchives/DASFNClient.scala

```diff
@@ -34,9 +44,39 @@ trait DASFNClient[F[_]: Async]:
       enc: Encoder[T]
   ): F[StartExecutionResponse]
 
+  def listStepFunctions(stepFunctionArn: String, status: Status): F[List[String]]
+
+  def sendTaskSuccess(taskToken: String): F[Unit]
+
 object DASFNClient:
 
+  extension [F[_]: Async, T](completableFuture: CompletableFuture[T])
+    private def liftF: F[T] = Async[F].fromCompletableFuture(Async[F].pure(completableFuture))
+
+  enum Status(val sdkStatus: String):
```

Comment:
Should we add all of [the statuses](https://docs.aws.amazon.com/step-functions/latest/apireference/API_DescribeExecution.html#StepFunctions-DescribeExecution-response-status)?

Reply:
Yeah we should, good idea. If we get one in one of the missing ones it will fail.